iceoryx_hoofs/test/moduletests/test_cxx_expected.cpp

```diff
@@ -578,4 +578,77 @@ TEST_F(expected_test, MoveAssignmentIsNotEnforcedInMoveConstructor)
         ASSERT_THAT(destination.has_error(), Eq(true));
     }
 }
+
+TEST_F(expected_test, TwoErrorOnlyExpectedWithEqualErrorAreEqual)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "471b406d-8dd3-4b82-9d46-00c21d257461");
+    auto sut1 = expected<TestError>::create_error(TestError::ERROR1);
+    auto sut2 = expected<TestError>::create_error(TestError::ERROR1);
+    EXPECT_TRUE(sut1 == sut2);
+    EXPECT_FALSE(sut1 != sut2);
+}
+
+TEST_F(expected_test, TwoErrorOnlyExpectedWithUnequalErrorAreUnequal)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "bcc2f9f1-72a1-41ed-ac8a-2f48cdcfbc56");
+    auto sut1 = expected<TestError>::create_error(TestError::ERROR1);
+    auto sut2 = expected<TestError>::create_error(TestError::ERROR2);
+    EXPECT_FALSE(sut1 == sut2);
+    EXPECT_TRUE(sut1 != sut2);
+}
+
+TEST_F(expected_test, TwoErrorOnlyExpectedWithValuesAreEqual)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "75b25c16-fb79-4589-ab0f-bc73bb9fc2bb");
+    auto sut1 = expected<TestError>::create_value();
+    auto sut2 = expected<TestError>::create_value();
+    EXPECT_TRUE(sut1 == sut2);
+    EXPECT_FALSE(sut1 != sut2);
+}
+
+TEST_F(expected_test, TwoErrorOnlyExpectedWithErrorAndValueAreUnequal)
```

Comment:
Could you add the same test for the expected which can contain an error and success value?